src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -19,6 +27,7 @@
 public class NameNodeExecutor extends AbstractNodeExecutor {
```

Comment:
With the health checks running, I'm not convinced that you have to modify the logic in NameNodeExecutor this much. The current code was created and works well with the state machine and is much simpler and less complex than this logic. You should be able to do something similar to what is currently in the health check branch, i.e.

healthCheckJN = new TimedHealthCheck(driver, journalNodeTask);
      timer.scheduleAtFixedRate(healthCheckJN, 30000, 60000);

You should be then only adding a few lines of health checks to this class.